roguelike-gruid-project/internal/ui/tiles_test.go

```
@@ -0,0 +1,139 @@
+	// Test that floor tiles have variety (multiple options)
+	floorTiles := tm.GetAvailableTiles('.')
+	if len(floorTiles) < 2 {
+		t.Error("Floor tiles should have multiple variants for variety")
+	}
+	
+	// Test that all returned paths are properly formed
+	for _, tile := range floorTiles {
+		if !filepath.IsAbs(tile) {
+			t.Errorf("Tile path should be absolute: %s", tile)
+		}
+		if !filepath.HasPrefix(tile, tempDir) {
```

golangci-lint (1.64.8)

SA1019: filepath.HasPrefix has been deprecated since Go 1.0 because it shouldn't be used: HasPrefix does not respect path boundaries and does not ignore case when required. 

(staticcheck)